tastypie/authorization.py

```diff
@@ -263,3 +257,88 @@ def delete_detail(self, object_list, bundle):
             raise Unauthorized("You are not allowed to access that resource.")
 
         return True
+
+
+class DjangoObjectAuthorization(Authorization):
+    '''
+    Uses permission checking from ``django.contrib.auth`` to map
+    ``POST / PUT / DELETE / PATCH`` to their equivalent Django auth
+    permissions.
+
+    Both the list & detail simply check the object they're based on.
+    Object level authorization api is added since django 1.5. However,
+    it will default to no-access unless an AUTHENTICATION_BACKENDS is
+    setup to handle those checks.
+    '''
+
+    # By default, follows `ModelAdmin` "convention" to use `app.change_model`
+    # `django.contrib.auth.models.Permission` for both viewing and updating.
+    # https://docs.djangoproject.com/es/1.9/topics/auth/default/#permissions-and-authorization
+    READ_PERM_CODE = getattr(settings, 'TASTYPIE_READ_PERM_CODE', 'change')
```

Comment:
Yeah, that'd be great, more code reuse is better. Just make sure to:
1. Change READ_PERM_CODE = getattr(settings, 'TASTYPIE_READ_PERM_CODE', 'change') to READ_PERM_CODE = 'change'
2. Add tests for DjangoObjectAuthorization.
3. Fix tests for DjangoAuthorization (verify that reading without change permissions fails, verify that reading with change permissions succeeds)
4. I think the current docs warn that DjangoAuthorization always allows read permissions, that should be updated.
5. Add docs for DjangoObjectAuthorization.
6. Mention this PR in docs/release_notes.dev.rst